[Doc Feature] [Apps in shared channels - Notification for indirect membership update and notification for team (host/non-host)[4532183]

v-bijinc opened the pull request:
Apps in shared channels - updated content for notification for indirect membership and notification for team (host/non-host)

v-bijinc committed:
Update shared-channels.md

Updated content for indirect membership for shared channels

v-bijinc committed:
Update shared-channels.md

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/b78c8516d6976b84a1a2938a93c8a3317f59cbbd/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 77 | 41 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/5ff1a9ae-c1d7-4ff2-a383-78181dd841f4) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Update shared-channels.md

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/14240ed88fae5f78e255b1a7c15c828791a81fec/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 77 | 40 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/1529917a-5dd5-4474-864a-884e55f458fe) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Update shared-channels.md

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/92bc6a252d4046e05c67f076115ab20188811f81/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 81 | 36 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/846d0cc4-74be-40f2-92b2-b2081d17a9a3) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Update shared-channels.md

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/de0eaadd2c6ec2e9518782be2507c614ff3b8eba/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 84 | 33 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/65c5e56f-57c5-477e-a596-fc309fee0424) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Update shared-channels.md

v-bijinc committed:
Merge branch 'main' into notifications-for-shared-channels

v-bijinc committed:
Update shared-channels.md

v-bijinc committed:
Merge branch 'main' into notifications-for-shared-channels

v-bijinc committed:
Merge branch 'main' into notifications-for-shared-channels

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/ba247312158f0876ec04ecc08f763669dcd2382d/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 84 | 33 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/f97a7601-0dca-4609-8734-d036d829df21) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Update shared-channels.md

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/828b89af402e0e6433adf786bd8dae1659e6760b/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 83 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/d17ccbb0-5da9-45bc-b771-768581355617) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Update shared-channels.md

v-bijinc committed:
Merge branch 'main' into notifications-for-shared-channels

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/74da053eed75f6e2b2f9bdeb3e8233aba0ad3606/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 83 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/bd03d7ac-e49e-48a4-b78a-8fc90bf2ecf5) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Updates to notifications for indirect membership

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/TOC.yml](https://github.com/MicrosoftDocs/msteams-docs/blob/158d22ef134206d77ef13ace36fa84cd296b0819/msteams-platform%2FTOC.yml) | 91 | 27 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/02ba39c8-cfc9-4932-b3ca-27aa105b7621) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/158d22ef134206d77ef13ace36fa84cd296b0819/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 83 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/4b762462-35d7-4ff1-b4ce-e79e214b9e66) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/whats-new.md](https://github.com/MicrosoftDocs/msteams-docs/blob/158d22ef134206d77ef13ace36fa84cd296b0819/msteams-platform%2Fwhats-new.md) | 90 | 108 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/56ac0821-e5e6-4b65-86cb-86ff2c8a3bbe) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Update whats-new.md

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/TOC.yml](https://github.com/MicrosoftDocs/msteams-docs/blob/d55c73407865b534d00085d36ae70a419cda3019/msteams-platform%2FTOC.yml) | 91 | 27 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/3c4c05af-ee7f-471c-9327-dd4bb1c13d25) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/d55c73407865b534d00085d36ae70a419cda3019/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 83 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/534995ff-1681-4e7d-b192-45ea25bae5d6) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/whats-new.md](https://github.com/MicrosoftDocs/msteams-docs/blob/d55c73407865b534d00085d36ae70a419cda3019/msteams-platform%2Fwhats-new.md) | 90 | 108 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/01c6564d-317b-4e2b-9440-940737f188f3) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Update whats-new.md

v-preethah reviewed (dismissed):
Approved.

v-bijinc committed:
Update shared-channels.md

v-bijinc committed:
Update shared-channels.md

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/TOC.yml](https://github.com/MicrosoftDocs/msteams-docs/blob/20919116c94fafe730a2d3d215cb25d21dea8cc9/msteams-platform%2FTOC.yml) | 91 | 27 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/003714fe-5031-452b-b1ea-72b0e54dd8cd) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/20919116c94fafe730a2d3d215cb25d21dea8cc9/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 83 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/37c51dd4-211a-43b5-9fa2-5cf1e60d81bd) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/whats-new.md](https://github.com/MicrosoftDocs/msteams-docs/blob/20919116c94fafe730a2d3d215cb25d21dea8cc9/msteams-platform%2Fwhats-new.md) | 90 | 108 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/96faed9b-b8df-41cb-9f31-385c66efabe9) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Merge branch 'main' into notifications-for-shared-channels

v-bijinc committed:
Update shared-channels.md

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/TOC.yml](https://github.com/MicrosoftDocs/msteams-docs/blob/f095a4c6b41f660109b0a5ada85a8c6c35609048/msteams-platform%2FTOC.yml) | 91 | 27 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/083d2d05-0684-4894-b600-79110c97129c) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/f095a4c6b41f660109b0a5ada85a8c6c35609048/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 83 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/30195f67-26cc-4fc9-8fd4-3fe16f8dfeaf) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/whats-new.md](https://github.com/MicrosoftDocs/msteams-docs/blob/f095a4c6b41f660109b0a5ada85a8c6c35609048/msteams-platform%2Fwhats-new.md) | 90 | 108 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/95704084-1e91-4af7-b0d1-5902cd8f84bd) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Updates to content

acrolinxatmsft1 commented:
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/TOC.yml](https://github.com/MicrosoftDocs/msteams-docs/blob/ac4b8ba635b27e94cf0a81ba7d8d27d0abdc1a11/msteams-platform%2FTOC.yml) | 91 | 27 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/8b45233b-ad1c-45a4-ab67-55eafa6de59e) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/concepts/build-and-test/shared-channels.md](https://github.com/MicrosoftDocs/msteams-docs/blob/ac4b8ba635b27e94cf0a81ba7d8d27d0abdc1a11/msteams-platform%2Fconcepts%2Fbuild-and-test%2Fshared-channels.md) | 83 | 34 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/9b2d1066-06cf-4b6d-ab4d-2efd106b56e3) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/whats-new.md](https://github.com/MicrosoftDocs/msteams-docs/blob/ac4b8ba635b27e94cf0a81ba7d8d27d0abdc1a11/msteams-platform%2Fwhats-new.md) | 90 | 112 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/d63ee01e-7efc-4a1f-b83c-3e6f6fe6c699) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

v-bijinc committed:
Update shared-channels.md

v-bijinc committed:
Update shared-channels.md

SatyaReddy-MSFT reviewed (approved):
Approved

NehaHEDAU-MSFT committed:
Updated new subscription resources

NehaHEDAU-MSFT committed:
Merge branch 'main' into notifications-for-shared-channels

NehaHEDAU-MSFT committed:
added spaces

NehaHEDAU-MSFT committed:
Changed the content for sharedWithTeams notifications

NehaHEDAU-MSFT committed:
Changed the subscription resource

NehaHEDAU-MSFT committed:
changed heading level

ChetanSharma-msft commented on msteams-platform/whats-new.md:
```diff
@@ -36,13 +36,9 @@ Teams platform features that are available to all app developers.</br>
-* ***May 19, 2025***: </br> Branding updates: </br> • Teams Toolkit is now Microsoft 365 Agents Toolkit. </br> • Teams App Test Tool is now Microsoft 365 Agents Playground. </br> • Teams App CLI, TeamFx CLI, and Teams Toolkit CLI are now Microsoft 365 Agents Toolkit CLI. </br> • `teamsapp` CLI signature is now referred as `atk`.
-* ***May 16, 2025***: [Use Meeting AI Insights API to fetch conversation summaries, action items, and mentions directly from Teams meetings.](graph-api/meeting-transcripts/meeting-insights.md)
-* ***May 13, 2025***: [Introducing agent usage analytics for third-party apps in Developer Portal](/microsoftteams/platform/concepts/build-and-test/analyze-your-apps-usage-in-developer-portal?tabs=thirdpartyapps).
+***June 27, 2025***: [Apps can now recieve notificarions when users are added to or removed from a shared channel through assoicated teams, helping developers maintain accurate access control](/microsoftteams/platform/concepts/build-and-test/shared-channels)
```
Check if 27th June is valid date.

NehaHEDAU-MSFT replied:
Have not made changes to this block at all.

ChetanSharma-msft replied:
Update as per feature release date:

***August 15 2025***

NehaHEDAU-MSFT replied:
done

ChetanSharma-msft commented on msteams-platform/whats-new.md:
```diff
@@ -56,6 +52,11 @@ Teams platform features that are available to all app developers.</br>
 | **Date** | **Update** | **Find here** |
 | -------- | --------- | ----------------|
+| 29/05/2025 | Introducing app manifest v1.22 with support for custom activity icons and prefetch of nested app authentication tokens.| [Microsoft 365 app manifest schema reference.](/microsoft-365/extensibility/schema/?view=m365-app-1.22&preserve-view=true) |
+| 19/05/2025 | Branding updates: </br> • Teams Toolkit is now Microsoft 365 Agents Toolkit </br> • Teams App Test Tool is now Microsoft 365 Agents Playground </br> • Teams App CLI, TeamFx CLI, and Teams Toolkit CLI are now Microsoft 365 Agents Toolkit CLI </br> • `teamsapp` CLI signature is now `atk` | [Microsoft 365 Agent Toolkit](/microsoft-365/developer/overview-m365-agents-toolkit?toc=%2Fmicr ...
```
Are these changes related to vNext?

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -73,12 +73,72 @@ You can get direct shared channel membership by using the `hostTeamGroupID` from
+To subscribe to indirect membership updates, use the following Microsoft Graph resource URL:
+
+`/teams/{team-id}/channels/getAllMembers?notifyOnIndirectMembershipUpdate=true&suppressWhenSharedUnsharedWithTeam=true`
+
```
Added the documentation for the new subscription that supports /shared with teams. We have tested and added it to the sample code.

NehaHEDAU-MSFT committed:
Merge branch 'main' into notifications-for-shared-channels

ChetanSharma-msft committed:
Update whats-new.md

Not part of feature.

ChetanSharma-msft commented on msteams-platform/whats-new.md:
```diff
@@ -36,13 +36,9 @@ Teams platform features that are available to all app developers.</br>
 * ***July 10, 2025***: [Custom icons for Teams notifications help your app stand out and increase engagement through more personalized activity feed updates.](/graph/teams-send-activityfeednotifications#custom-activity-icons-in-activity-feed-notifications)</br>
 
-**2025 May**
+**2025 June**
```
Should it be August?

NehaHEDAU-MSFT committed:
Changed Date and removed spelling mistake

NehaHEDAU-MSFT committed:
Updated content for all members

Added a hyperlink for allMembers API and tabulated the difference between direct and indirect members.

NehaHEDAU-MSFT committed:
Fixed Hyperlink issue and added a Note for Duplicates

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
````diff
@@ -73,12 +73,80 @@ You can get direct shared channel membership by using the `hostTeamGroupID` from
     ```http
-    GET /teams/{host-team-group-id}/channels/{channel-id}/sharedWithTeams/{teamX}/members
+    GET /teams/{host-team-group-id}/channels/{channel-id}/sharedWithTeams/{sharewithteamsId}/allowedMembers
     ```
````
Added how developers can identify direct and indirect member of a shared channel using the annotation property. Also added a note for duplicates. Please check if the intended meaning is preserved.

NehaHEDAU-MSFT committed:
changes to include content for direct and indirect membership updates

also changed query parameters.

NehaHEDAU-MSFT committed:
changed content for sharedwith Teams Notifications

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -73,12 +73,80 @@ You can get direct shared channel membership by using the `hostTeamGroupID` from
+
+* Use `allMembers` API to retrieve the complete updated list of current members.
+* Use `allowedMembers` to retrieve only newly added members during shared events.
+* Avoid using `DoesUserHaveAccessAsync` for every user unless necessary.
```
removed as information already covered.

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -73,12 +73,80 @@ You can get direct shared channel membership by using the `hostTeamGroupID` from
+
+### Handle bulk membership changes
+
+In large-scale scenarios, such as sharing a channel with a large team or removing multiple users, Teams suppresses individual notifications to reduce volume.
```
changed heading and rephrased the content

NehaHEDAU-MSFT committed:
corrected errors from Acrolinx

NehaHEDAU-MSFT committed:
minor corrections

sumanac commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -59,6 +62,26 @@ You can collaborate with external members outside of your organization using sha
+> [!NOTE]
+> You might receive duplicate notifications when a member is added to a shared channel. This scenario can happen if the member is already part of the shared channel directly or through another linked team.
+>
+> To avoid duplicate notifications:
```
I don't think it can be avoided. But More like identifying if this is a duplicate or not and take appropriate action.

sumanac commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -59,6 +62,26 @@ You can collaborate with external members outside of your organization using sha
+> * Compare incoming indirect membership notifications against the cached list.
+> * Ignore the notification if the member already exists; either directly or indirectly.
+
 You can get direct shared channel membership by using the `hostTeamGroupID` from `getContext` and following these steps:
```
Let's combine this section and next section.

sumanac commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -73,12 +96,75 @@ You can get direct shared channel membership by using the `hostTeamGroupID` from
+
+To reduce notification overload during membership updates, such as when a shared channel is added to or removed from a team with thousands of members, use the new SharedWithTeams subscription resource:
+
+`/teams/{team-id}/channels/{channel-id}/sharedWithTeams`
```
Let's also provide updating the list of all members for a shared channel using /allmembers api after receiving shared with or unshared from a team notification.

NehaHEDAU-MSFT committed:
Added definition for direct and indirect members

NehaHEDAU-MSFT committed:
made corrections to the text as per changes suggested

NehaHEDAU-MSFT committed:
Update shared-channels.md

NehaHEDAU-MSFT committed:
Merge branch 'main' into notifications-for-shared-channels

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -21,13 +21,13 @@ Microsoft Teams Connect shared channels allow members of a channel to collaborat
 
-Teams Connect shared channels facilitate secure collaboration seamlessly. Allow external users outside of your organization to collaborate with internal users in Teams without changing their user context. Enhance user experience unlike using guest accounts, for example, the members must sign out of Teams and sign in again using a guest account. Teams applications extend the powerful collaboration ...
+Teams connects shared channels, which facilitates secure collaboration seamlessly. Allow external users outside of your organization to collaborate with internal users in Teams without changing their user context. Enhance user experience unlike using guest accounts, for example, the members must sign out of Teams and sign in again using a guest account. Teams applications extend the powerful coll ...
 
```
changed and added other benefits.

NehaHEDAU-MSFT committed:
part 3 changes incorporated

NehaHEDAU-MSFT committed:
Merge branch 'main' into notifications-for-shared-channels

NehaHEDAU-MSFT committed:
changed as per Microsoft style guide

NehaHEDAU-MSFT commented:
@microsoft-github-policy-service agree company="Microsoft"

NehaHEDAU-MSFT committed:
changed  content under  in-tenant or out-tenant

NehaHEDAU-MSFT committed:
Acrolinx changes incorporated

NehaHEDAU-MSFT committed:
updated Intro and performed self-review

NehaHEDAU-MSFT committed:
updated changes by Dev0

NehaHEDAU-MSFT committed:
Update shared-channels.md

Pawank-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -1,5 +1,5 @@
 ---
-title: Teams Connect Shared Channels
+title: Teams Connects Shared Channels
```
"Teams Connects Shared Channels" should be "Teams Connect Shared Channels"
Please confirm with Sandip and if needed change in all places.

NehaHEDAU-MSFT replied:
done

Pawank-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -8,26 +8,32 @@ ms.topic: conceptual
+
+Shared channels ensure:
+
+* Enhanced user experience through cross-tenanants collaboration
```
"cross-tenanants" → "cross-tenants"

Pawank-MSFT commented on msteams-platform/whats-new.md:
```diff
@@ -44,13 +44,7 @@ Teams platform features that are available to all app developers.</br>
-* ***May 19, 2025***: </br> Branding updates: </br> • Teams Toolkit is now Microsoft 365 Agents Toolkit. </br> • Teams App Test Tool is now Microsoft 365 Agents Playground. </br> • Teams App CLI, TeamFx CLI, and Teams Toolkit CLI are now Microsoft 365 Agents Toolkit CLI. </br> • `teamsapp` CLI signature is now referred as `atk`.
-* ***May 16, 2025***: [Use Meeting AI Insights API to fetch conversation summaries, action items, and mentions directly from Teams meetings.](graph-api/meeting-transcripts/meeting-insights.md)
-* ***May 13, 2025***: [Introducing agent usage analytics for third-party apps in Developer Portal](/microsoftteams/platform/concepts/build-and-test/analyze-your-apps-usage-in-developer-portal?tabs=thirdpartyapps).
+***Aug 15, 2025***: [Apps can now recieve notifications when users are added to or removed from a shared channel through assoicated teams, helping developers maintain accurate access control](/microsoftteams/platform/concepts/build-and-test/shared-channels)
```
"recieve" → "receive"

Pawank-MSFT replied:
"assoicated" → "associated"

Pawank-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -8,26 +8,32 @@ ms.topic: conceptual
 ## Enable your app for shared channels
 
-SupportedChannelTypes is an optional property that enables your app in non-standard channels. If your app supports the team scope and the property is defined, Teams enables your app in each channel type accordingly. Private and shared channels are supported. For more information, see [supportedChannelTypes](../../resources/schema/manifest-schema.md#supportedchanneltypes).
+SupportedChannelTypes is an optional property that enables your app in nonstandard channels. If your app supports the team scope and the property is defined, Teams enables your app in each channel type accordingly. Private and shared channels are supported. For more information, see [supportedChannelTypes](../../resources/schema/manifest-schema.md#supportedchanneltypes).
```
"nonstandard" → "non-standard"

Pawank-MSFT reviewed (commented):
please fix some typo issue

NehaHEDAU-MSFT committed:
fixed internal review changes, typos and what's new section changes

NehaHEDAU-MSFT committed:
added a bullet in what's new section

NehaHEDAU-MSFT committed:
updated property names as per UI

sumanac reviewed (changes requested):
Thanks for working on it.

NehaHEDAU-MSFT committed:
changed content under validate user access for membership updates

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -36,15 +42,9 @@ SupportedChannelTypes is an optional property that enables your app in non-stand
     ]
```
yes in new manifest versions  supportedChannelFeatures will be rolled out, and supportedChannelTypes will not be available.
 
supportedChannelTypes will continue to be in the versions before this feature. However, since the future release cycle is TBD, we will document that after release.

NehaHEDAU-MSFT committed:
Update shared-channels.md

NehaHEDAU-MSFT committed:
Merge branch 'main' into notifications-for-shared-channels

NehaHEDAU-MSFT committed:
Update shared-channels.md

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -60,39 +60,112 @@ You can collaborate with external members outside of your organization using sha
+<a name='azure-ad-native-identity'></a> and determine if the member is an in-tenant or out-tenant.
 
 ## Microsoft Entra native identity
 
```
removed it

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -60,39 +60,112 @@ You can collaborate with external members outside of your organization using sha
+3. Compare the `TenantId` of the member to the `ownerTenantId` property
+<a name='azure-ad-native-identity'></a> and determine if the member is an in-tenant or out-tenant.
 
 ## Microsoft Entra native identity
```
removed it, after confirming with Gaurav

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -60,39 +60,112 @@ You can collaborate with external members outside of your organization using sha
+
+`/teams/{team-id}/channels/{channel-id}/sharedWithTeams`
+
+The sharedWithTeams subscription sends a single notification when a channel is shared or unshared with a team. It avoids thousands of per-user notifications and improves performance for apps that monitor membership changes. Ensure that you update the shared channel member list using the /allMembers API after receiving a 'shared with' or 'unshared from' team notification.
```
done

NehaHEDAU-MSFT committed:
incorporated changes

NehaHEDAU-MSFT committed:
updated APIs

NehaHEDAU-MSFT commented on msteams-platform/concepts/build-and-test/shared-channels.md:
```diff
@@ -60,39 +60,112 @@ You can collaborate with external members outside of your organization using sha
+> * Access membership data (both direct and indirect members).
+> * Receive and respond to membership change notifications.
+
+### Manage indirect membership in shared channels
```
added to the list

NehaHEDAU-MSFT committed:
fixed PR Warning

NehaHEDAU-MSFT committed:
deleted note

SirajShaik-MSFT committed:
Merge branch 'main' into notifications-for-shared-channels

SirajShaik-MSFT commented:
@microsoft-github-policy-service agree company="Microsoft"

NehaHEDAU-MSFT committed:
Acrolinx

NehaHEDAU-MSFT committed:
Merge branch 'notifications-for-shared-channels' of https://github.com/MicrosoftDocs/msteams-docs into notifications-for-shared-channels

v-shalinirob commented:
@microsoft-github-policy-service agree

NehaHEDAU-MSFT committed:
Update shared-channels.md

NehaHEDAU-MSFT committed:
removed test

ChetanSharma-msft commented:
> @v-bijinc please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement
> ### Contribution License Agreement
> This Contribution License Agreement (**“Agreement”**) is agreed to by the party signing below (**“You”**), and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your contributions to Microsoft open source projects. This Agreement is effective as of the latest signature date below.
> 
> 1. **Definitions**.
>    **“Code”** means the computer software code, whether in human-readable or machine-executable form,
>    that is delivered by You to Microsoft under this Agreement.
>    **“Project”** means any of the projects owned or managed by Microsoft and offered under a license
>    approved by the Open Source Initiative ([www.opensource.org](http://www.opensource.org)).
>    **“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any
>    Project, including but not limited to communication on electronic mailing lists, source code control
>    systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
>    discussing and improving that Project, but excluding communication that is conspicuously marked or
>    otherwise designated in writing by You as “Not a Submission.”
>    **“Submission”** means the Code and any other copyrightable material Submitted by You, including any
>    associated comments and documentation.
> 2. **Your Submission**. You must agree to the terms of this Agreement before making a Submission to any
>    Project. This Agreement covers any and all Submissions that You, now or in the future (except as
>    described in Section 4 below), Submit to any Project.
> 3. **Originality of Work**. You represent that each of Your Submissions is entirely Your original work.
>    Should You wish to Submit materials that are not Your original work, You may Submit them separately
>    to the Project if You (a) retain all copyright and license information that was in the materials as You
>    received them, (b) in the description accompanying Your Submission, include the phrase “Submission
>    containing materials of a third party:” followed by the names of the third party and any licenses or other
>    restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
>    guidelines concerning Submissions.
> 4. **Your Employer**. References to “employer” in this Agreement include Your employer or anyone else
>    for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
>    Submission is made in the course of Your work for an employer or Your employer has intellectual
>    property rights in Your Submission by contract or applicable law, You must secure permission from Your
>    employer to make the Submission before signing this Agreement. In that case, the term “You” in this
>    Agreement will refer to You and the employer collectively. If You change employers in the future and
>    desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
>    and secure permission from the new employer before Submitting those Submissions.
> 5. **Licenses**.
> 
> * **Copyright License**. You grant Microsoft, and those who receive the Submission directly or
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
>   Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
>   the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
>   parties.
> * **Patent License**. You grant Microsoft, and those who receive the Submission directly or
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
>   Your patent claims that are necessarily infringed by the Submission or the combination of the
>   Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
>   import or otherwise dispose of the Submission alone or with the Project.
> * **Other Rights Reserved**. Each party reserves all rights not expressly granted in this Agreement.
>   No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
>   granted by implication, exhaustion, estoppel or otherwise.
> 
> 6. **Representations and Warranties**. You represent that You are legally entitled to grant the above
>    licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
>    have disclosed under Section 3). You represent that You have secured permission from Your employer to
>    make the Submission in cases where Your Submission is made in the course of Your work for Your
>    employer or Your employer has intellectual property rights in Your Submission by contract or applicable
>    law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
>    have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
>    You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
>    REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
>    EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
>    PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
>    NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
> 7. **Notice to Microsoft**. You agree to notify Microsoft in writing of any facts or circumstances of which
>    You later become aware that would make Your representations in this Agreement inaccurate in any
>    respect.
> 8. **Information about Submissions**. You agree that contributions to Projects and information about
>    contributions may be maintained indefinitely and disclosed publicly, including Your name and other
>    information that You submit with Your Submission.
> 9. **Governing Law/Jurisdiction**. This Agreement is governed by the laws of the State of Washington, and
>    the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
>    Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
>    exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
>    defenses of lack of personal jurisdiction and forum non-conveniens.
> 10. **Entire Agreement/Assignment**. This Agreement is the entire agreement between the parties, and
>     supersedes any and all prior agreements, understandings or communications, written or oral, between
>     the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

@microsoft-github-policy-service agree company="Microsoft"

NehaHEDAU-MSFT committed:
Update shared-channels.md

ChetanSharma-msft commented:
> @v-bijinc please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

NehaHEDAU-MSFT committed:
Update shared-channels.md

NehaHEDAU-MSFT committed:
Merge branch 'main' into notifications-for-shared-channels
